statsmodels/genmod/tests/test_glm.py

```diff
@@ -855,42 +855,85 @@ def test_gradient_irls():
-        rslt_gradient = mod_gradient.fit(start_params=rslt_irls.params, method="newton")
-
-        assert_allclose(rslt_gradient.params,
-                        rslt_irls.params, rtol=1e-6, atol=1e-6,
-                        err_msg=family_name + " params")
-
-        gradient_bse = rslt_gradient.bse
-        if family_name == "negbinom":
-            ehess = mod_gradient.hessian(rslt_gradient.params, observed=False)
-            gradient_bse = np.sqrt(-np.diag(np.linalg.inv(ehess)))
-        assert_allclose(gradient_bse, rslt_irls.bse, rtol=1e-6,
-                        atol=1e-6, err_msg=family_name + " SE")
 
+    for family_class, family_links in families:
+
+       for link in family_links:
+           print((family_class, link))
+           if (family_class, link) == (fam.Poisson, lnk.identity):
+               lin_pred = 20 + exog.sum(1)
+           elif (family_class, link) == (fam.Binomial, lnk.log):
+               lin_pred = -1 + exog.sum(1) / 8
+           elif (family_class, link) == (fam.Poisson, lnk.sqrt):
+               lin_pred = 2 + exog.sum(1)
+           elif (family_class, link) == (fam.InverseGaussian, lnk.log):
+               lin_pred = -1 + exog.sum(1)
+           elif (family_class, link) == (fam.InverseGaussian, lnk.identity):
+               lin_pred = 20 + 5*exog.sum(1)
+               lin_pred = np.clip(lin_pred, 1e-4, np.inf)
+           elif (family_class, link) == (fam.InverseGaussian, lnk.inverse_squared):
+               lin_pred = 0.5 + exog.sum(1) / 5
+               continue
+           elif (family_class, link) == (fam.InverseGaussian, lnk.inverse_power):
+               lin_pred = 1 + exog.sum(1) / 5
+           elif (family_class, link) == (fam.NegativeBinomial, lnk.identity):
+               lin_pred = 20 + 5*exog.sum(1)
+               lin_pred = np.clip(lin_pred, 1e-4, np.inf)
+           elif (family_class, link) == (fam.NegativeBinomial, lnk.inverse_squared):
+               lin_pred = 0.1 + np.random.uniform(size=exog.shape[0])**4
+               continue
+           elif (family_class, link) == (fam.NegativeBinomial, lnk.inverse_power):
+               lin_pred = 1 + exog.sum(1) / 5
+           else:
+               lin_pred = np.random.uniform(size=exog.shape[0])
+
+           endog = gen_endog(lin_pred, family_class, link)
+
+           mod_irls = sm.GLM(endog, exog, family=family_class(link=link))
+           rslt_irls = mod_irls.fit(method="IRLS")
+
+           mod_gradient = sm.GLM(endog, exog, family=family_class(link=link))
+           rslt_gradient = mod_gradient.fit(start_params=rslt_irls.params, method="newton")
+
+           assert_allclose(rslt_gradient.params,
+                           rslt_irls.params, rtol=1e-6, atol=1e-6,
+                           err_msg=str(family_class) + " params")
+
+           assert_allclose(rslt_gradient.llf, rslt_irls.llf,
+                           rtol=1e-6, atol=1e-6, err_msg=str(family_class) + " params")
```

Comment:
is `" params"` copy paste without changing the name?
Wouldn't the error message would already show `assert_allclose(rslt_gradient.llf, rslt_irls.llf,` with the attribute name?